impeller/entity/contents/filters/gaussian_blur_filter_contents_unittests.cc

```diff
@@ -487,5 +487,138 @@ TEST(GaussianBlurFilterContentsTest, Coefficients) {
+              },
+          },
+  };
+
+  KernelPipeline::FragmentShader::KernelSamples fast_kernel_samples =
+      LerpHackKernelSamples(kernel_samples);
+  EXPECT_EQ(fast_kernel_samples.sample_count, 3);
+
+  KernelPipeline::FragmentShader::KernelSample* samples =
+      kernel_samples.samples;
+  KernelPipeline::FragmentShader::KernelSample* fast_samples =
+      fast_kernel_samples.samples;
+
+  //////////////////////////////////////////////////////////////////////////////
+  // Check output kernel.
+
+  EXPECT_FLOAT_EQ(fast_samples[0].uv_offset.x, -1.3333333);
+  EXPECT_FLOAT_EQ(fast_samples[0].uv_offset.y, 0);
+  EXPECT_FLOAT_EQ(fast_samples[0].coefficient, 0.3);
+  EXPECT_FLOAT_EQ(fast_samples[1].uv_offset.x, 0);
+  EXPECT_FLOAT_EQ(fast_samples[1].uv_offset.y, 0);
+  EXPECT_FLOAT_EQ(fast_samples[1].coefficient, 0.4);
+  EXPECT_FLOAT_EQ(fast_samples[2].uv_offset.x, 1.3333333);
+  EXPECT_FLOAT_EQ(fast_samples[2].uv_offset.y, 0);
+  EXPECT_FLOAT_EQ(fast_samples[2].coefficient, 0.3);
+
+  //////////////////////////////////////////////////////////////////////////////
+  // Check output of fast kernel versus original kernel.
+
+  Scalar data[5] = {0.25, 0.5, 0.5, 1.0, 0.2};
+  Scalar original_output =
+      samples[0].coefficient * data[0] + samples[1].coefficient * data[1] +
+      samples[2].coefficient * data[2] + samples[3].coefficient * data[3] +
+      samples[4].coefficient * data[4];
+
+  auto lerp = [](const Point& point, Scalar left, Scalar right) {
+    Scalar int_part;
+    Scalar fract = fabsf(modf(point.x, &int_part));
+    if (point.x < 0) {
+      return left * fract + right * (1.0 - fract);
+    } else {
+      return left * (1.0 - fract) + right * fract;
+    }
+  };
+  Scalar fast_output =
+      /*1st*/ lerp(fast_samples[0].uv_offset, data[0], data[1]) *
+          fast_samples[0].coefficient +
+      /*2nd*/ data[2] * fast_samples[1].coefficient +
+      /*3rd*/ lerp(fast_samples[2].uv_offset, data[3], data[4]) *
+          fast_samples[2].coefficient;
+
+  EXPECT_NEAR(original_output, fast_output, 0.01);
+}
+
+TEST(GaussianBlurFilterContentsTest, LerpHackKernelSamplesComplex) {
+  Scalar sigma = 10.0f;
+  int32_t blur_radius = static_cast<int32_t>(
+      std::ceil(GaussianBlurFilterContents::CalculateBlurRadius(sigma)));
+  BlurParameters parameters = {.blur_uv_offset = Point(1, 0),
+                               .blur_sigma = sigma,
+                               .blur_radius = blur_radius,
+                               .step_size = 1};
+  KernelPipeline::FragmentShader::KernelSamples kernel_samples =
+      GenerateBlurInfo(parameters);
+  EXPECT_EQ(kernel_samples.sample_count, 33);
+  KernelPipeline::FragmentShader::KernelSamples fast_kernel_samples =
+      LerpHackKernelSamples(kernel_samples);
+  EXPECT_EQ(fast_kernel_samples.sample_count, 17);
+  float data[33];
+  srand(0);
+  for (int i = 0; i < 33; i++) {
+    data[i] = 255.0 * static_cast<double>(arc4random()) / RAND_MAX;
```

Comment:
I usually ignore the arc4random lint because I don't think its available on every platform we build on.

Reply:
Oh man, that is annoying, it got me twice.